up-l2/dispatchers/README.adoc

```diff
@@ -123,3 +123,17 @@ When a dispatcher is unable to dispatch an event for a given reason (queue full,
 .RPC Error Flow
 [#rpc-error-flow]
 image::rpc_flow.png[RPC Error Handling]
+
+== File Publish Handling
+
+When a uStreamer instance dispatcher is supporting file transfer, the file details will generally be passed by reference, and the transfer will be queued for delivery.  uP traffic priorities will affect when content can be delivered, and file size(s) along with available link bandwidth will also affect transfer duration.
+
+Sustained integrity of files queued for transfer is essential, and the providing uEs will need to prevent the files from being changed/deleted until such time as their transfer has been completed.   
+
+The basic Publish, and uStreaming logic is a fire-and-forget approach, with content queued and delivered but no handshaking between the subscribing uEs.  This works well for content passed by value, but cases where content is passed by reference and the transfer can take an indeterminate duration to complete necessitate communication with the Publishing uE.  Dispatchers and uTransport instances are not exposed services and lack visible protobufs.  Intent for Notification is for a "known" status response to be provided to the Publisher for the type of content being published, delivered as generic ".any" protobuf within a Notification message.   The uProtocol_options.protobuf will provide an optional notification request field to trigger notifications to the publisher by applicable dispatcher(s) - in this case the File Streaming dispatcher, to indicate completion of transfer along with success/failure details.
```

Comment:
Then files need to be attached to a request/response and NOT to a notification if you want to be notified when the upload is complete. That or both sides publish the file is being sent and then publish the file has been consumed. What you are doing with notification is fundamentally changing the design pattern.